server/server.rs

```diff
@@ -215,62 +255,128 @@ impl Server {
         }
+        let authenticator =
+            grpc::authenticator::Authenticator::new(credential_verifier, token_manager, diagnostics_manager);
 
-        let cert_path = encryption_config.cert.as_ref().ok_or_else(|| ServerOpenError::MissingTLSCertificate {})?;
-        let cert = fs::read_to_string(cert_path).map_err(|source| ServerOpenError::CouldNotReadTLSCertificate {
-            path: cert_path.display().to_string(),
-            source: Arc::new(source),
-        })?;
-        let cert_key_path =
-            encryption_config.cert_key.as_ref().ok_or_else(|| ServerOpenError::MissingTLSCertificateKey {})?;
-        let cert_key =
-            fs::read_to_string(cert_key_path).map_err(|source| ServerOpenError::CouldNotReadTLSCertificateKey {
-                path: cert_key_path.display().to_string(),
-                source: Arc::new(source),
-            })?;
-        let mut tls_config = ServerTlsConfig::new().identity(Identity::from_pem(cert, cert_key));
+        grpc_server
+            .layer(&authenticator)
+            .add_service(typedb_protocol::type_db_server::TypeDbServer::new(service))
+            .serve_with_shutdown(address, async {
+                // The tonic server starts a shutdown process when this closure execution finishes
+                shutdown_receiver.changed().await.expect("Expected shutdown receiver signal");
+            })
+            .await
+            .map_err(|source| ServerOpenError::GRPCServe { address, source: Arc::new(source) })
+    }
 
-        if let Some(root_ca_path) = &encryption_config.root_ca {
-            let root_ca = fs::read_to_string(root_ca_path).map_err(|source| ServerOpenError::CouldNotReadRootCA {
-                path: root_ca_path.display().to_string(),
-                source: Arc::new(source),
-            })?;
-            tls_config = tls_config.client_ca_root(Certificate::from_pem(root_ca)).client_auth_optional(true);
+    async fn serve_http(
+        address: SocketAddr,
+        credential_verifier: Arc<CredentialVerifier>,
+        token_manager: Arc<TokenManager>,
+        diagnostics_manager: Arc<DiagnosticsManager>,
+        encryption_config: &EncryptionConfig,
+        mut shutdown_receiver: tokio::sync::watch::Receiver<()>,
+        service: http::typedb_service::TypeDBService,
+    ) -> Result<(), ServerOpenError> {
+        let authenticator =
+            http::authenticator::Authenticator::new(credential_verifier, token_manager, diagnostics_manager);
+
+        let encryption_config = http::encryption::prepare_tls_config(encryption_config)?;
+        let http_service = Arc::new(service);
+        let router_service = http::typedb_service::TypeDBService::create_protected_router(http_service.clone())
+            .layer(authenticator)
+            .merge(http::typedb_service::TypeDBService::create_unprotected_router(http_service))
+            .into_make_service();
```

Comment:
Can we add this please. It lets my browser make requests from the web-app
```suggestion
        let router_service = http::typedb_service::TypeDBService::create_protected_router(http_service.clone())
            .layer(authenticator)
            .merge(http::typedb_service::TypeDBService::create_unprotected_router(http_service))
            .layer(tower_http::cors::CorsLayer::permissive())
            .into_make_service();
```

Reply:
I think we can actually do slightly better here - the official builds of Studio will only be deployed to `studio.typedb.com` and `studio.typedb.dev`, and of course locally run Studio typically ends up on `localhost`.

So if our CORS policy allows precisely these three domains, we get both a functioning Studio and the security benefits of CORS.

Reply:
Was thinking about this. Should we make it configurable? How likely is it that the user wants to connect DIRECTLY to the database from their website like we do in studio?

Reply:
Hmmm, that's a fair point - ultimately the server belongs to the user, so we might be overstepping our boundaries by imposing fixed restrictions on it